synapse/tests/utils.py

```diff
@@ -819,6 +819,10 @@ def jsonlines(self) -> typing.List[dict]:
         '''Get the messages as jsonlines. May throw Json errors if the captured stream is not jsonlines.'''
         return jsonlines(self.getvalue())
 
+    def __str__(self):
```

Comment:
Seeing how this is used; should we just add an `expect(self, valu: str)` API that checks if the `valu` is in the recorded buffer so far?